src/backends/gpu/kernels/reduce_by_key.hpp

```diff
@@ -8,6 +8,43 @@ namespace gpu {
 
 namespace impl{
 
+constexpr unsigned mask_all = 0xFFFFFFFF;
```

Comment:
Style question: would it be more readable to just have two regions in this file, a __CUDACC_VER_MAJOR < 9 and not with 3 definitions rather than this line-wise preprocessor testing?

Reply:
Not too big of a deal, either way.
Two regions would make for 5 additional lines of code (remove 6, add 11), because the function prototypes would have to be replicated, which makes more room for cut and paste error.
This approach also makes it very clear that the only thing that varies is the intrinsic.